sympy/core/mul.py

```diff
@@ -863,6 +863,9 @@ def as_real_imag(self, deep=True, **hints):
                 else:
                     if a.is_Add:
                         addterms *= a
+                    elif a.is_Pow and a.base.is_Add and not a.has(re, im):
+                        r, i = a.as_real_imag()
+                        other.append(r + i*S.ImaginaryUnit)
```

Comment:
`a.as_real_imag()` is already computed above. The values computed there can be reused.